### PR TITLE
Include the macOS tarballs in `versions.json`, and add an `extension` field that gives the file extension for a given file

### LINKED ISSUE #3: Add the `.tar.gz` files for macOS
Currently, it looks like `versions.json` only has the `.dmg` files for macOS. Can we also add the `.tar.gz` files?

If I recall correctly, we should have the `.tar.gz` files available for macOS for all released versions of Julia. @staticfloat is this correct?

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #3

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Okay, it looks like we are missing `x86_64-apple-darwin` tarballs for 1.1, 1.2, and 1.3.

```julia
julia> for arch in ["x86_64", "aarch64"]
           println(arch)
           println.([string(i) * ". " * version for (i, version) in enumerate(sort!([file.version for file in filter!(file -> file.arch == arch && file.extension == "tar.gz" && file.os == "mac", vcat([v.files for v in values(JSON3.read(read("versions.json", String)))]...))]))]);
       end
x86_64
1. 1.0.5
2. 1.4.1
3. 1.4.2
4. 1.5.0
5. 1.5.0-beta1
6. 1.5.0-rc2
7. 1.5.1
8. 1.5.2
9. 1.5.3
10. 1.6.0
11. 1.6.0-beta1
12. 1.6.0-rc1
13. 1.6.0-rc2
14. 1.6.0-rc3
15. 1.6.1
16. 1.6.2
17. 1.6.3
18. 1.6.4
19. 1.6.5
20. 1.6.6
21. 1.6.7
22. 1.7.0
23. 1.7.0-beta1
24. 1.7.0-beta2
25. 1.7.0-beta3
26. 1.7.0-beta4
27. 1.7.0-rc1
28. 1.7.0-rc2
29. 1.7.0-rc3
30. 1.7.1
31. 1.7.2
32. 1.7.3
33. 1.8.0-beta1
34. 1.8.0-beta2
35. 1.8.0-beta3
36. 1.8.0-rc1
37. 1.8.0-rc3
aarch64
1. 1.7.0
2. 1.7.0-beta3
3. 1.7.0-beta4
4. 1.7.0-rc1
5. 1.7.0-rc2
6. 1.7.0-rc3
7. 1.7.1
8. 1.7.2
9. 1.8.0-beta1
10. 1.8.0-beta3
11. 1.8.0-rc1
12. 1.8.0-rc3

julia>
```

**Comment by DilumAluthge** [comment]
@SaschaMann @staticfloat This is ready for review.

As mentioned above, we do not have `x86_64-apple-darwin` tarballs for 1.1, 1.2, or 1.3. However, that does not block this PR. If we add the aforementioned tarballs after this PR is merged, we can just trigger a new `workflow_dispatch` run.

**Comment by DilumAluthge** [comment]
Bump @SaschaMann @staticfloat

**Comment by DilumAluthge** [comment]
Bump @SaschaMann @staticfloat

File: .github/workflows/CI.yml
Comment (SaschaMann) [inline .github/workflows/CI.yml:16]: We don't need all of this in this repo given the low activity. Same with the timeouts. Could you either remove the changes or move them to a separate commit to keep the history cleaner, please?

File: schema.json
Comment (SaschaMann) [inline schema.json:140]: This info is already part of the file name, what's the motivation behind a new field? I'm not opposed to it but want to clarify the reason.

If we want to tell apart the different kinds of archives, perhaps a mime type field is a better solution?

File: src/VersionsJSONUtil.jl
Comment (SaschaMann) [inline src/VersionsJSONUtil.jl:222]: The reordering adds noise to the commit, could you perhaps move it to its own commit?

**Review by SaschaMann — COMMENTED** [review]
Some minor comments, otherwise this looks good

**Comment by SaschaMann** [comment]
> Okay, it looks like we are missing x86_64-apple-darwin tarballs for 1.1, 1.2, and 1.3.

Would it be possible to add these? I don't know how much effort it takes, but it would allow us to ditch the `dmg` files entirely in setup-julia and other tooling, and only use the more convenient tarballs.

**Comment by davidanthoff** [comment]
> Would it be possible to add these?

That would also fix https://github.com/JuliaLang/juliaup/issues/184 and https://github.com/JuliaLang/juliaup/issues/183 :)

File: .github/workflows/CI.yml
Comment (DilumAluthge) [reply .github/workflows/CI.yml:16]: Moved to #5

File: schema.json
Comment (DilumAluthge) [reply schema.json:140]: Yeah, the motivation is to be able to distinguish different archives (e.g. `.tar.gz` versus `.zip`) without the consumer needing to manually parse the filename. 

I'm not sure that MIME type is the best format. E.g. the MIME type for `.dmg` is just `application/octet-stream`, which doesn't seem super useful.

**Comment by DilumAluthge** [comment]
I think Elliot has a script somewhere for repackaging DMGs as tarballs. Ideally, we can have him run the script on *every* single release of Julia. It will take a long time, but it only needs to be done once.

File: src/VersionsJSONUtil.jl
Comment (DilumAluthge) [reply src/VersionsJSONUtil.jl]: I've removed the reordering.

File: src/VersionsJSONUtil.jl
Comment (DilumAluthge) [reply src/VersionsJSONUtil.jl]: No specific reason, just personal preference. I've switched back to the list of platforsms.

File: src/VersionsJSONUtil.jl
Comment (SaschaMann) [inline src/VersionsJSONUtil.jl:234]: This feels a bit pointless since we know and expect a large number of version combinations to not be available, so the exact number doesn't mean much. But I don't mind it being there either.

**Review by SaschaMann — APPROVED** [review]

File: schema.json
Comment (SaschaMann) [reply schema.json:140]: > E.g. the MIME type for .dmg is just application/octet-stream, which doesn't seem super useful.

I don't like dmg files :D 

In that case, I'm not sure if it's more efficient to include it in the json file vs parsing it on each run but let's add the key.

**Comment by SaschaMann** [comment]
> I think Elliot has a script somewhere for repackaging DMGs as tarballs. Ideally, we can have him run the script on every single release of Julia. It will take a long time, but it only needs to be done once.

I don't care much for <1.0 versions, but all of 1.x would be great.

**Pull request merged**
